sklearn/utils/tests/test_stats.py

```diff
@@ -96,3 +96,71 @@ def test_weighted_percentile_2d():
         _weighted_percentile(x_2d[:, i], w_2d[:, i]) for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
+
+
+def test_weighted_percentile_nan_filtered():
+    """Test that calling _weighted_percentile on an array with nan values returns
+    the same results as calling _weighted_percentile on a filtered version of the data.
+    We test both with sample_weight of the same shape as the data and for
+    one-dimensional sample_weight."""
+
+    rng = np.random.RandomState(42)
+    array = rng.rand(10, 100)
+
+    nan_array = array.copy()
```

Comment:
I have renamed `nan_array` into `array_with_nans`, but have not gotten rid of the original `array`, because it is used in the test to have its filtered version's percentiles compared with the `array_with_nans` percentiles. Am I missing something?